tests/phpunit/tests/test-wpml-elementor-translate-ids.php

```diff
@@ -399,4 +399,46 @@ public function it_should_translate_location_condition_any_child_of() {
 		$this->assertEquals( $translate_id,
 			$subject->translate_location_condition_sub_id( (string) $sub_id, $parsed_conditions ) );
 	}
+
+	/**
+	 * @test
+	 * @group wpmlcore-7714
+	 */
+	public function it_translates_product_ids() {
+		$post_id       = 123;
+		$product_id    = 456;
+		$tr_product_id = 789;
+
+		$data = [ [
+			'id' => '4a3eaf7',
```

Comment:
@dgwatkins, I would use a 'wc-add-to-cart' inside another widget so we can actually test the recursive part of the method (otherwise tests will pass even if I remove `$data['elements'] = $this->translate_product_ids( $data['elements'], $post_id );`.

Reply:
@strategio added one more level. Now we have section > column > widget.